src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -242,8 +261,22 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
     this.setState({ showUpdatePasswordModal: true });
   };
 
-  updatePassword = (passwords: UpdatePasswordFormType) => {
-    // TODO: update password when API is ready
+  /* Update password */
+  updatePassword = (password: string) => {
+    const { title, description, auth } = this.props.existingDataSource;
+    const updateAttributes: DataSourceAttributes = {
+      title,
+      description,
+      endpoint: undefined,
```

Comment:
why use undefined instead of `existingDataSource.endpoint`?

Reply:
nope, API call will fail if we send endpoint in payload .. it has to be undefined as the user should not be able to update the endpoint once created